homeworkW11D02/src/ba/bitcamp/homeworkW11D02/Client.java

```diff
@@ -0,0 +1,220 @@
+			JScrollPane.VERTICAL_SCROLLBAR_AS_NEEDED,
+			JScrollPane.HORIZONTAL_SCROLLBAR_AS_NEEDED);
+
+	// Regular socket
+	private Socket client;
+
+	// List of directories
+	private ArrayList<String> listOfDirectories;
+
+	/**
+	 * Constructor for creating Client chat
+	 */
+	public Client() {
+
+		/*
+		 * If client (user) types ENTER key on the keyboard and if the text
+		 * field is not empty, written message will be sent to a server
+		 */
+		textField.addKeyListener(new KeyAdapter() {
+			@Override
+			public void keyPressed(KeyEvent e) {
+				if (e.getKeyCode() == KeyEvent.VK_ENTER
+						&& !textField.getText().equals("")) {
+					try {
+
+						BufferedWriter writer = new BufferedWriter(
+								new OutputStreamWriter(client.getOutputStream()));
+						writer.write("Client: " + textField.getText());
+						writer.newLine();
+						writer.flush();
+						textArea.append("Client: " + textField.getText() + "\n");
+						textArea.setCaretPosition(textArea.getDocument()
+								.getLength());
+						textField.setText("");
+					} catch (IOException e1) {
+						e1.printStackTrace();
+					}
+				}
+			}
+		});
+
+		/*
+		 * Configuring and adding components to lower panel
+		 */
+		downPanel.setBorder(BorderFactory.createTitledBorder("Write Message"));
+		downPanel.setLayout(new BorderLayout());
+		downPanel.add(textField, BorderLayout.CENTER);
+
+		/*
+		 * Configuring and adding components to upper panel
+		 */
+		upPanel.setBorder(BorderFactory.createTitledBorder("Chat room"));
+		upPanel.setLayout(new BorderLayout());
+		upPanel.add(scrollPane, BorderLayout.CENTER);
+
+		/*
+		 * Configuring text area
+		 */
+		textArea.setLineWrap(true);
+		textArea.setEditable(false);
+		textArea.setForeground(Color.RED);
+		textArea.setBackground(Color.LIGHT_GRAY);
+		textArea.setFont(new Font("Serif", Font.BOLD, 15));
+
+		/*
+		 * Configuring and adding components to main panel
+		 */
+		mainPanel.setLayout(new BorderLayout());
+		mainPanel.add(upPanel, BorderLayout.CENTER);
+		mainPanel.add(downPanel, BorderLayout.SOUTH);
+
+		// Adding main panel to the window
+		add(mainPanel);
+
+		/*
+		 * Configuring window
+		 */
+		setTitle("Client chat");
+		setSize(450, 600);
+		setDefaultCloseOperation(JFrame.EXIT_ON_CLOSE);
+		setResizable(false);
+		setLocationRelativeTo(null);
+		setVisible(true);
+
+		/*
+		 * Starting client
+		 */
+		try {
+			client = new Socket("localhost", 6699);
+
+			/*
+			 * Reading messages from server
+			 */
+			BufferedReader reader = new BufferedReader(new InputStreamReader(
+					client.getInputStream()));
+			boolean end = false;
+			while (!end) {
+				String readText = "";
+				String line = reader.readLine();
+
+				/*
+				 * If message starts with "/list" followed with file path, it is
+				 * command to find all subdirectories in certain directory on
+				 * client's PC and put them in to a list. After that list shows
+				 * in both server's and client's chat room.
+				 */
+				if (line.indexOf("/list") != -1) {
```

Comment:
A good way to implement this would be to have an event parser that is implemented elsewhere and just used here.

```
EventType eventType = EventParser.parse(string);

EventHandler.handle(eventType, params);
```